checks.d/ceph.py

```diff
@@ -0,0 +1,137 @@
+"""ceph check
+Collects metrics from ceph clusters
+"""
+
+# stdlib
+import os
+import json
+
+# project
+from checks import AgentCheck
+from utils.subprocess_output import get_subprocess_output
+from config import _is_affirmative
+
+NAMESPACE = "ceph"
+
+class Ceph(AgentCheck):
+    """ Collect metrics and events from ceph """
+
+    DEFAULT_CEPH_CMD = '/usr/bin/ceph'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+    def _run_command(self, cmd):
+        fd = os.popen(cmd)
+        raw = fd.read()
+        fd.close()
+        return json.loads(raw)
+
+    def _collect_raw(self, ceph_cmd, instance):
+        use_sudo = _is_affirmative(instance.get('use_sudo', False))
+        if use_sudo:
+            test_sudo = os.system('setsid sudo -l < /dev/null')
```

Comment:
We should ensure that the user has enough permissions to run the ceph command, and if not write a clear error message on how to fix that.